debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/structureformat/extensions/J9ObjectStructureFormatter.java

```diff
@@ -248,53 +263,110 @@ private void formatArrayObject(PrintStream out, J9ClassPointer localClazz, U8Poi
 		}
 	}
 	
-	private void formatStringObject(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject) throws CorruptDataException 
+	private void formatStringObject(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject, long address) throws CorruptDataException 
 	{
 		out.println(String.format("J9VMJavaLangString at %s {\n", localObject.getHexAddress()));
-		printJ9ObjectFields(out, tabLevel, localClazz, dataStart, localObject);
+		printJ9ObjectFields(out, tabLevel, localClazz, dataStart, localObject, address);
 		padding(out, tabLevel);
 		out.println(String.format("\"%s\"\n", J9ObjectHelper.stringValue(localObject)));
-		out.println("}");		
+		out.println("}");
+	}
+	
+	public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject, long address) throws CorruptDataException 
+	{
+		printJ9ObjectFields(out, tabLevel, localClazz, dataStart, localObject, address, null, false);
 	}
 	
-	private void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject) throws CorruptDataException 
+	public static void printJ9ObjectFields(PrintStream out, int tabLevel, J9ClassPointer localClazz, U8Pointer dataStart, J9ObjectPointer localObject, long address, String[] nestingHeirarchy, boolean showNestedFields) throws CorruptDataException 
```

Comment:
Why must this be `public`? This will extend the public DTFJ API which we need to do carefully.

Reply:
This issue has just been moved to `PrintObjectFieldsHelper`, but the question remains.